terraform_compliance/extensions/terraform.py

```diff
@@ -345,7 +345,7 @@ def _mount_references(self):
                 if ref_list:
                     ref_type = self.configuration['resources'][resource]['expressions'].get('type', {})
 
-                    if not ref_type and not self.is_type(resource, 'data'):
+                    if (not ref_type or not isinstance(ref_type, str)) and not self.is_type(resource, 'data'):
```

Comment:
hmm so this will run for `int`, `bool`, `list` and `dict` as long a resource type is not `data` ?